### PR TITLE
Refactor damage model

### LINKED ISSUE #319: Improve the broken wing model
@wlbragg what do you think if we improve the broken wing model by showing the wing structure through the broken point? I am talking about these green metal structures: http://www.mikejamesmedia.com/modo_media/modo_wip/cessna_310k/cessna_310k_138_ribs.jpg

Here you can see how they look in the 172: http://4.bp.blogspot.com/-FbGczdVEcts/UQaMdCgj4cI/AAAAAAAAAAs/IquO8W5p6xc/s1600/Cessna+172++glass+cockpit+%287%29.jpg

If you like the idea, I can create a texture for that. If not, feel free to close this issue.

### LINKED ISSUE #592: Duplicate wing objects
With the current Blender files we have 10 objects for 10 states.
A lot of objects are partly duplicated because of this (the vertices).

Example 1: the middle section of `wing_1BB` and `wing_1BD` are (basically) the same.
Example 2: right outer wing part of `wing_1RGLB` and `wing_1LD` are the same.

| Left | Right | Blender |
| --- | --- | --- |
| ok | ok | (in default Blender file) |
| dmg | ok | wing_1LD |
| brk | ok | wing_1RGLB |
|  |  |  |
| ok | dmg | wing_1RD |
| dmg | dmg | wing_1BD |
| brk | dmg | wing_1RDLB |
|  |  |  |
| ok | brk | wing_1GRB |
| dmg | brk | wing_1LDRB |
| brk | brk | wing_1BB |
|  |  |  |
| col | col | wing1Collapse |

If we split the wings in two halves (left and right) we can reduce it to
6 or 7 objects for 10 states. If you want to make a change to one of the parts,
for example the left outer damaged wing, you only have to modify 1
object (`left_dmg`) instead of 3 (`wing_1LD`, `wing_1BD`, and `wing_1LDRB`).
It will make the model XML file easier and smaller.

These are the objects you get:
- `left_ok`
- `left_dmg`
- `right_ok`
- `right_dmg`
- `col_and_col` (or you can choose `left_col` + `right_col`)
- `inner_wing`

So now we only need about 5 select animations (`inner_wing` object is always
displayed) instead of 10.

The only thing the 3D modeller has to be aware of is to make sure the vertices
of the outer parts precisely align with the `inner_wing` object so that the user
doesn't see the cutting edge when a wing halve (left and/or right) is ok.

For all the 10 states here's which objects to display:

| Left | Right | Model XML file |
| --- | --- | --- |
| ok | ok | inner_wing + left_ok + right_ok |
| dmg | ok | inner_wing + left_dmg + right_ok |
| brk | ok | inner_wing + right_ok |
|  |  |  |
| ok | dmg | inner_wing + left_ok + right_dmg |
| dmg | dmg | inner_wing + left_dmg + right_dmg |
| brk | dmg | inner_wing + right_dmg |
|  |  |  |
| ok | brk | inner_wing + left_ok + right_brk |
| dmg | brk | inner_wing + left_dmg + right_brk |
| brk | brk | inner_wing |
|  |  |  |
| col | col | col_and_col or inner_wing + left_col + right_col |

### LINKED ISSUE #895: 3D model optimizations
I've been reading the mailing list a bit, and it seems we should apply some optimizations.

1. We can start with #592. So you get something like this:
    
    fgfs-wings
    
   All the objects can be in the same `c172p.blend` file and same textures can be used.
    
    I can do all the XML stuff in `Models/c172p.xml`.

2. We should use a as few textures as possible. We should also reuse UV coordinates if an object (and its texture) is symmetrical. For example, take a look at `fuselage.png`: apart from the baggage door, the right side of the fuselage is the same as the left side, so we can just use one side. Same for the propeller (just use one half) and that thing above the wheel. Same thing with `tail.png`, `wing.png`, and `floats.png`.
    
    `interior_windows.png`, `lights.png`, `panel_parts.png`, `prop.png` are very small textures, and should be added to to the other textures.
    
    Same thing with many instruments; they often use many small textures instead of just one big texture.

3. There are some unnecessary objects: https://github.com/c172p-team/c172p-detailed/blob/master/Models/c172p.xml#L3117-L3164 This can be one object called 'FuelSelector` with one rotate animation instead of four select animations.
    
    We can remove https://github.com/c172p-team/c172p-detailed/blob/master/Models/c172p.xml#L3505-L3551 if we decide to drop Rembrandt lighting.

### LINKED ISSUE #1027: Improve 3D model of the top of the nose on the left side
The 3D model of the top of the nose has a strange shape on the left side and is a bit buggy, see: 

image

### LINKED ISSUE #1029: wing mapping on CAP livery is broken
The UV-mapping  of the top side of the wings of the CAP livery is utterly broken since we added the wing edge object and remapped the wing. There are two possible solutions to this, both with negative sides:

* attempt to re-UV-map the top of the wings, risking to affect the other liveries, all of which look fine
* remove the CAP livery and leave things as they are. This also means that anyone attempting to create a new livery in the future with a fancy top wing design will be very frustrated. That said, we have some solid HD liveries at the moment.

I am fine leaving things as they are, since I suspect the pain/gain ratio is really high on this one. @wlbragg any thoughts?

### LINKED ISSUE #1031: Model the strobe light
I noticed that we have a navigation light in the 3D model, but no strobe light model:
image

as compared to 

image

The procedural light should also be moved to match the model's new position. I'd also appreciate some reference images ;)

Of course, you could always model these:
image

😁

### PR DESCRIPTION
**Pull request by wlbragg**
Fixes #319 
Fixes #592 
Fixes #895 
Fixes #1027
Fixes #1029 
Fixes #1031
:warning: Not ready to merge yet :warning:

### CODE REVIEW

**Comment by wlbragg** [comment]
@onox I inadvertently have the material settings change for the red tint in the prop commit in this, How can I back that out and leave the other commit. or can I?

Also, I messed up the previous wing split. The center section didn't extend to the wing struts where it needed to be so I am redoing it in this branch. We can merge this new split when we finish the damage model s and merge it all together.

**Comment by onox** [comment]
If you want to delete the first commit then you have to do `git rebase -i HEAD~1` and delete the commit that you want to  remove.

**Comment by wlbragg** [comment]
That is all local though, right? How do I go about changing it here in github, just push it again after I do what you said?

**Comment by gilbertohasnofb** [comment]
@wlbragg Have you seen what @onox proposed here https://github.com/c172p-team/c172p-detailed/pull/1018#issuecomment-374304596 ? We can just close that PR and work on the propeller here as well, what do you think?

**Comment by wlbragg** [comment]
Yeah, I just read it. OK, lets close the other PR and delete the branch, we'll work from here.

**Comment by gilbertohasnofb** [comment]
And in any case, here is the new texture for propeller disk blur for the default livery named `PropBlur.png`. Left is the front of the disk, right is the back. Once this is mapped I will go about making new ones for each livery, as I need to see it in the sim in order to make them. What do you think?

propblur

**Comment by gilbertohasnofb** [comment]
By the way, this is the same texture as the one used by the c182s, I just removed the colouring of the edges since the 4X-CHV has a black propeller both in the front and in the back.

**Comment by wlbragg** [comment]
>Once this is mapped I will go about making new ones for each livery, as I need to see it in the sim in order to make them. What do you think?

Yeah, are you going to push the textures here?

**Comment by gilbertohasnofb** [comment]
> Yeah, are you going to push the textures here?

Doing it right now, just wasn't sure of what you'd decide for the PRs.

**Comment by wlbragg** [comment]
@gilbertohasnofb  @onox 

>By the way, this is the same texture as the one used by the c182s,

You may have missed my post somewhere (maybe I managed to not post it even) about the darkness of the prop blur in the c182s. What I tried to convey is that I like our prop blur lighting as far as darkness or contrast much better than that of the c182s. I really hope at the end of all this prop refactoring we don't end up with a darker blur. I am not sure if that is entirely up to the texture or the transparency or both. If you agree, please help keep that in mind as we work through this.

**Comment by gilbertohasnofb** [comment]
@wlbragg there are some issues with the new propeller disk:

* it's not livery dependent at the moment
* it does not match the size of our spinner:

image

image

**Comment by wlbragg** [comment]
@onox 

Do you know why we are using this condition, around line 6596 in Models/c172p.xml?

~~~
<animation>
        <type>select</type>
        <object-name>landinglightcover</object-name>
        <condition>
            <not>
                <property alias="/params/lighting/landing/property"/>
            </not>
        </condition>
    </animation>
~~~
To save you from having to look it up
alias="/params/lighting/landing/property"
is
~~~
                <int n="4" alias="/sim/model/c172p/lighting/taxi"/>
                <int n="5" alias="/sim/model/c172p/lighting/landing"/>
~~~

I don't know why it is there and I can't think of any reason for it. We take care of conditions for the landinglightcover's in the damage model code section.

Speaking of landinglight covers, the problem of it crashing the sim has gone away so I am including the cover now. We can decide later if we want to break it during a damage event.

**Comment by gilbertohasnofb** [comment]
@wlbragg I just pushed a prop disk texture for the OH-CTL. If you implement that, I will use your work as template and do the same for all other liveries. Is that all right?

I also added a white background to the reflection maps as you asked, see if that works for you.

I will now add the broken wing texture to all wing.png files.

**Comment by gilbertohasnofb** [comment]
@wlbragg done, all wings have the broken model. I have nothing else to commit until you implement the propeller disk dependency to the liveries, which is when I will create the prop disks for all liveries and use your xml code for all liveries.

**Comment by wlbragg** [comment]
OK, I'm done also. Still to do in this PR is decide on whether to use lights or not during damage condition and verify all damage conditions are working correctly.

**Comment by gilbertohasnofb** [comment]
@wlbragg Great. I need to leave now but I will be back in a couple of hours and then I should get the propellers for all liveries up and working. Many thanks!

**Comment by gilbertohasnofb** [comment]
@wlbragg just did a quick test, that is looking _really_ good! :smile:

**Comment by wlbragg** [comment]
Thanks for the help, that really pushed this PR forward!

**Comment by onox** [comment]
> I don't know why it is there and I can't think of any reason for it. We take care of conditions for the landinglightcover's in the damage model code section.

If it is redundant, then you can remove it.

**Comment by wlbragg** [comment]
Left to do...

Decide what to do about lights during damage condition.
Verify all damage models are in sync with all damage conditions.
Add any unique prop blur liveries.

**Comment by onox** [comment]
@wlbragg See my comments in #592

**Comment by onox** [comment]
I think we can leave the lights on when damaged, but off when broken or collapsed.

**Comment by onox** [comment]
@dany93 Wing animations should be fixed now.

@wlbragg I've pushed a commit.

Should we still add rotate animations for `leftaileron_C` and/or `leftaileron_D` (and for the right aileron as well)?

**Comment by gilbertohasnofb** [comment]
> I think @gilbertohasnofb will take care of 3.

I just got back home, going to work on this still tonight. Then my part of this PR will be done. I let you guys know once I am ready to push.

**Comment by gilbertohasnofb** [comment]
@wlbragg I just pushed all propeller liveries and their respective xml. It's all looking really good! But I need you to take a look at two things:

* the propeller texture is in a 512 x 256 image, so that each side is 256 x 256. I used that as reference when making each livery match their propeller. But it turns out you did not map the disk to the full 256 x 256 area, but rather to a smaller subset. This means that all disks I created are 'cut short' in the sim, not matching their propellers. Do you think you could re-UV-map it? Just use OH-CTL as reference. E.g. notice how the texture has two white bands but the sim shows only one:

image

image

* now that the propeller disk is not fully black, the yellow and red colouring at sunset when at rainy or overcast light is even more dramatic. Do you have any idea why this is still plaguing our beautiful new propellers? I don't see this happening in other GA aircraft.

image

From my side, I am done and I will move back to working on a better engine sound in a different branch. If you guys need anything from me other than testing just give me a shout!

**Comment by wlbragg** [comment]
>Do you think you could re-UV-map it? Just use OH-CTL as reference. E.g. notice how the texture has two white bands but the sim shows only one:
Done.
prop

@onox even though you verified the damage animation I am also doing that now. I just want to see for myself the state of all of them. Plus it will help me visualize what left to do with any lighting and the ailerons or flaps.
I am so glad you took care of the damage logic as that part of it really messes with my brain, thanks for doing that.

**Comment by wlbragg** [comment]
OK, damage logic all looks good.

>I think we can leave the lights on when damaged, but off when broken or collapsed.

I agree.

>Should we still add rotate animations for leftaileron_C and/or leftaileron_D (and for the right aileron as well)?

Not when their collapsed for sure. The only time that happens is when you hit the ground hard. So on crash, as in collapse, ailerons are broken. But if damaged, yes they should be able to rotate. This is done!

Working on lights now.

**Comment by gilbertohasnofb** [comment]
> Done.

Looks really good, thanks for taking a look at this. So my side of this PR seems to be done.

**Comment by wlbragg** [comment]
>Looks really good, thanks for taking a look at this.

No problem, I still have to investigate the low light coloring after I finish the damage light logic. @onox did the damage nav left/right translation thankfully, but there is still the condition logic as to when to show the different lights. I have the landing and taxi lights finished. I changed the taxi/landing lights to break only when the left wing is collapsed (and gone of course (broken). But if it is damaged I now keep the taxi and landing lights I just translate them into the new position.

I also allow for flaps always and ailerons even if wings are damaged, not collapsed or broken (gone) of course.

**Comment by wlbragg** [comment]
@onox,
I had to add some individual logic for the left and right nav spotlights in order to get them to work independent of each other.
There were missing condition logic for some nav light damage conditions and tax/landing light damage conditions (ALS specific such as landing taxi from interior view or exterior view).

Also all the flap and aileron damage condition animations are accounted for and implemented.

So, all nav, taxi and landing light lighting conditions accounted for. All control surface damage animations accounted for.

Strobes are all that is left. The state of that is not working.
I set up some translations for the position of the default strobe objects VS using separate but additional strobe objects for the damaged positions. For some reason the translations aren't working as I expected. Not even the default. So I either have something wrong or you can't translate them like I am trying to do.

If you can please take a look and see if you can identify my problem.

Either that or I can set it back up like it was , but I really would like to optimize using translations instead of 4 additional strobe objects like it was.

File: c172p-set.xml [hunk 1]
```diff
@@ -385,9 +385,9 @@ http://forum.flightgear.org/viewtopic.php?f=4&t=25157
                 <path>Aircraft/c172p/Systems/engine.xml</path>
             </property-rule>
 
-            <property-rule n="104">
+            <!--property-rule n="104">
```
Comment (onox) [inline c172p-set.xml]: Don't you need to re-enable this?

**Comment by wlbragg** [comment]
Oops, will do that now.

**Comment by onox** [comment]
What's not working about the strobes? Translations would indeed be better than extra objects.

**Comment by wlbragg** [comment]
They won't position according to the translation I assign. It's like it breaks even the default translation position. The default strobes and the damage conditioned positions are all the same and too far away from the wing tips.

You should be able to tell easy enough what is wrong visually and what I was trying to o.

Anyway, I am done for the night. It got really late here. I'll check back in later today, in about 8-10 hours.
See ya!

**Comment by onox** [comment]
@wlbragg `<offset-m>` might need to be negative instead of positive (when experimenting with `strobe1` for the left wing). But even then they were not fully in the correct position. Why don't you move the strobe objects in `c172p.blend` to where they should be when the wings are ok?

**Comment by onox** [comment]
Also, there are still some old `strobe*` objects in the XML file on several lines.

**Comment by wkitty42** [comment]
On 03/24/2018 04:52 PM, wlbragg wrote:
> Left to do...
> 
> Decide what to do about lights during damage condition.


i know i'm late with my thoughts about this comment but how about using a random 
"coin flip" to determine if the lights are still on after the damage or not... 
you know? like in a real accident, sometimes the wires are still good and other 
times the wires will break... so just do a flip of the quarter to see if it is 
heads or tails and have the light on if it is heads...

**Comment by dany93** [comment]
> ....but how about using a random "coin flip" to determine if the lights are still on after the damage or not...

Very superfluous IMHO... And will probably be unnoticed.

**Comment by wkitty42** [comment]
On 03/25/2018 11:47 AM, dany93 wrote:
>     ....but how about using a random "coin flip" to determine if the lights are
>     still on after the damage or not...
> 
> Very superfluous IMHO... And will probably be unnoticed.


maybe... but imagine the surprise when it is noticed realism is greater with 
than without...

besides, it shouldn't be too much code (a couple of lines?) to choose TRUE or 
FALSE randomly... the idea could even be used in the model's code if a bulb 
burns out and needs to be changed... if random damage is enabled, of course ;)

**Comment by wlbragg** [comment]
>how about using a random "coin flip" 

I don't dislike the idea, but there are so many areas of code that control the different lighting it is more than I want to deal with (especially now that it is almost finished).
I didn't realize just how many types of lighting effects we deploy, it is substantial.

**Comment by wlbragg** [comment]
@onox why didn't we employ procedural strobe lights like we did nav?

**Comment by wlbragg** [comment]
@onox I'm done with the strobes. I ended up keeping a single duplicate of the strobes for damaged strobes. I couldn't figure out how to translate them correctly. I was trying to use a condition to select which translation to use and as soon as I coded the damage translation both translations broke. So I give up, it's not worth my time especially as the original works and we still reduced the footprint in the amount of strobe objects used overall.

So as far as I am concerned everything is finished for this PR with the exception of figuring out why the propblur takes on strange colors in low light.

**Comment by wlbragg** [comment]
>What's this Plane.014Xparams thing btw?

That was all stuff we inherited from the original.
Unless it is being used somewhere else it is kind of superfluous.

The params properties
~~~
<Plane.014Xparams>
    <light-near>0.8</light-near>
    <light-med>3.2</light-med>
    <light-far>160</light-far>
</Plane.014Xparams>
~~~
are created only to be used in the dist-scale animation
~~~
<animation>
    <type>dist-scale</type>
    <object-name>strobe1</object-name>
    <object-name>strobe2</object-name>
    <!-- Damage Mod-->
    <object-name>strobe1D</object-name>
    <object-name>strobe2D</object-name>
    <interpolation>
        <entry>
            <ind>0</ind>
            <dep alias="/Plane.014Xparams/light-near"/>
        </entry>
        <entry>
            <ind>500</ind>
            <dep alias="/Plane.014Xparams/light-med"/>
        </entry>
        <entry>
            <ind>16000</ind>
            <dep alias="/Plane.014Xparams/light-far"/>
        </entry>
    </interpolation>
</animation>
~~~
Same kind of thing was going on for the group name
~~~
<animation>
    <name>Plane.014XGroup</name>
    <object-name>strobe1</object-name>
    <object-name>strobe2</object-name>
    <!-- Damage Mod -->
    <object-name>strobe1BD</object-name>
    <object-name>strobe2BD</object-name>
    <object-name>strobe1Collapse</object-name>
    <object-name>strobe2Collapse</object-name>
</animation>
~~~
which was only being used for the alpha-test, 
~~~
<animation>
        <type>alpha-test</type>
        <object-name>Plane.014XGroup</object-name>
        <alpha-factor>0.001</alpha-factor>
    </animation>
~~~
which kind of made it superfluous so I removed it.

Do we want to just hard code the numbers into the dist-scale animation or continue to use the pre-fabricated properties?

**Comment by wlbragg** [comment]
@gilbertohasnofb I can't reproduce the low light red/yellow colored probblur. Mine looks correct.

**Comment by gilbertohasnofb** [comment]
@wlbragg would you try this: load the sim at SBTA, load AW as Low Pressure Region and advance the time to around 6:30pm. See if you can spot the same (OH-CTL is probably the best livery for this but all props do get red):

fg-1

**Comment by wlbragg** [comment]
OK, I see it.

**Comment by gilbertohasnofb** [comment]
@wlbragg I added "Fixes #771" to the first comment, since I understand you are planning to push your improved ac model here, right?

Also, one more bug report: there are some ugly collisions with the wing light from some specific angles:

image

**Comment by wlbragg** [comment]
Dang it, it appears we have a problem that has been introduced somewhere along the line.
fgfs-screen-006

**Comment by wlbragg** [comment]
Hum, maybe not. It does this in other aircraft. Part of it is the fur shader not playing nice with the reflection shader. The lighter shad of the side windows VS the front and back I think may be a reflection shader issue.

**Comment by wlbragg** [comment]
@gilbertohasnofb 

I mentioned at the start of the propeller modifications that we watch out and not get the propblur too dark. I really liked the shade or lightness our original blur, I think the c182s's are too dark. Maybe it's because they have a 3 blade prop. Anyway, I think ours got too dark, it looks fake and unnatural IMHO when they get too dark. So I lightened up the default and the OH-CTL prop.png to what I think is a better shade. I would even go lighter if it were up to me. But I can live with a happy medium.
Would you please take a look at what I did with those two props and see what you think. If your OK with the change maybe you could edit the rest of the props to be more in line with that lightness and contrast.

I adjusted the nav lights best I could. I think they are considerably better.

I guess I am done with everything else barring anything we missed.

**Comment by dany93** [comment]
@gilbertohasnofb wrote:
>.... not get the propblur too dark.......Anyway, I think ours got too dark

That was my feeling too.

**Comment by gilbertohasnofb** [comment]
> Dang it, it appears we have a problem that has been introduced somewhere along the line.

@wlbragg Oh, so you think those zigzag lines are due to the internal glass mesh? 

> I adjusted the nav lights best I could. I think they are considerably better.

Great, I will test them later.

> So I lightened up the default and the OH-CTL prop.png to what I think is a better shade. I would even go lighter if it were up to me. But I can live with a happy medium.

I see, and I agree with you and @dany93 on the issue. But it's not a matter of making them lighter but rather of making them more transparent by lightening up only the alpha channel. I haven't tested your modifications to see what you did but I will do so later today and then I will be glad to work on all others myself and push the modifications. Do notice though that when you change a png file you should also send a PR to the texture repository, as all xcf sources are there. But in this case you can just leave it as it is, I will reproduce your results for the default livery and then apply the same procedure to all other prop disks. Makes sense?

I am unfortunately super busy today but I should have some free time in the evening to tackle this.

**Comment by wlbragg** [comment]
>Oh, so you think those zigzag lines are due to the internal glass mesh?

I was more referring to the difference in brightness between the front and back glass reflection and the side glass reflection.

No time today either on my part either, later tonight.

**Comment by gilbertohasnofb** [comment]
> I was more referring to the difference in brightness between the front and back glass reflection and the side glass reflection.

Could that have been introduced by using a white background in the reflection maps? See commit https://github.com/c172p-team/c172p-detailed/pull/1019/commits/0c4e1d319e9c35c5837a0b8476afbeb3d4297db2

**Comment by gilbertohasnofb** [comment]
I had understood from the discussion here https://github.com/c172p-team/c172p-detailed/issues/1022#issuecomment-374808692 that we should change their background to white. If that's the cause of these issues we can just revert that commit of mine.

**Comment by gilbertohasnofb** [comment]
@wlbragg

* I pushed new prop disks which are more transparent. I did not make them as light as you did, I think you actually had overdone it (you could barely see the prop on the default livery). See what you think about this, but IMO it is looking good

fg-1
fg-3
fg-2

* I also pushed a small fix for the N35799 livery after a comment by lego
* the VHF antennas are not positioned very symmetrically, see image below. Red lines are the sides of the aircraft, notice how the distance is very different. Also, notice how the antennas align in different positions in relation the normal map lines (purple): 

fg-0

* the normal map of top side of the wing is not matching the fuel caps any longer (see the circled rivets). Is there room to improve that or is that too difficult given all recent modifications to the wing?  

fg

**Comment by legoboyvdlp** [comment]
If this is where we report exterior things,
Please note that some of these things might already be there - but based on memory, they aren't ;)

https://camo.githubusercontent.com/5b2f8a2a873e965a419554c39f0da37e1f85405b/687474703a2f2f63646e2d7777772e6169726c696e6572732e6e65742f70686f746f732f6169726c696e6572732f352f342f322f343030373234352e6a70673f763d763439663437353933623130

This 1980 Model P has an antenna under the tail, also seen at 

https://camo.githubusercontent.com/e7a55531c2f18fae1acc1e18feaf58d1aa65775d/687474703a2f2f63646e2d7777772e6169726c696e6572732e6e65742f70686f746f732f6169726c696e6572732f312f302f322f333938333230312e6a70673f763d763465373631383365643238

Also, on that photograph, the Cessna has static doschargers on the wingtip, I have seen photographs of the Cessna with and without them.

Also, there is a cowling handle on both sides of the fuselage - can't remember from memory if ours has it on the right side only, as I suspect. In the above image, that part is in shadow.

**Comment by gilbertohasnofb** [comment]
This is most definitely _not_ the place to report exterior things. This is the place where @wlbragg and I ask each other to do small tweaks that are easily done in a short amount of time, or report bugs with the current aircraft. Things like a new 3D model for an antenna which hasn't been modelled should have their own issue, which might get tackled on an already open PR (or not).

About the antenna under the tail, we used photos of the OH-CTL when we first started this project and that model most certainly doesn't have that antenna.

About the handle, you are right it should also be on the right side of the nose, @wlbragg would you care to duplicate it?

**Comment by legoboyvdlp** [comment]
Would you mind a link to the OHCTL gallery please? I searched for it for 30 mins, but was unable to find the new link... :(

**Comment by wkitty42** [comment]
On 03/26/2018 07:13 PM, Jonathan Redpath wrote:
> If this is where we report exterior things,
> Please note that some of these things might already be there - but based on 
> memory, they aren't ;)


ugh... yeah, make a new issue on this, please... or it may be possible to find 
an existing exterior defects thread? probably not but worth looking, anyway :)

**Comment by wkitty42** [comment]
On 03/26/2018 08:42 PM, Jonathan Redpath wrote:
> Would you mind a link to the OHCTL gallery please? I searched for it for 30 
> mins, but was unable to find the new link... :(


i don't think there's one specific to that craft... there is the liveries repo 
but that's only for livery stuff... not 3D model stuff...

**Comment by wlbragg** [comment]
>Could that have been introduced by using a white background in the reflection maps?

No that wasn't the cause. I tried a couple other aircraft and they all exhibited the same difference between the side glass and the front glass. In fact our front windshield is now one piece and even that one does the same thing, the side is brighter then the front.

Is everyone else seeing the same thing I am?

**Comment by wlbragg** [comment]
>See what you think about this, but IMO it is looking good

:+1: 

>About the handle, you are right it should also be on the right side of the nose, @wlbragg would you care to duplicate it?

:+1: 

>the VHF antennas are not positioned very symmetrically

:+1: 

>the normal map of top side of the wing is not matching the fuel caps any longer

Not sure when that happened. I wonder if it did that when we made the leading edge a separate object?

I'll see if I can clean it up.

**Comment by onox** [comment]
> Would you mind a link to the OHCTL gallery by tigert please? I searched for it for 30 mins, but was unable to find the new link... :(

Do you mean this comment https://github.com/c172p-team/c172p-detailed/issues/798#issuecomment-368190245?

**Comment by wlbragg** [comment]
OK, I finished everything that was brought to my attention. The only outstanding issue is the reflection maps and I really don't think there is anything we can do about them.

I'd like someone to verify that what is seen in the image I posted above is the same on other systems and that it is the same on other aircraft and not just ours. I'm talking about the reflection image in the glass being darker in to sloped front and back portions of the glass and lighter in the more vertical side glass.

If that is true for other aircraft then it is an effect issue and not specific to the c172p. I verified it to be the same in at east a couple other aircraft but would like a couple more opinions from other's observations.

**Comment by gilbertohasnofb** [comment]
>> Would you mind a link to the OHCTL gallery by tigert please? I searched for it for 30 mins, but was unable to find the new link... :(
>
> Do you mean the link in this comment #798 (comment)?

That's not it, @onox. I believe @tigert's gallery is not online any longer. I did save a couple of those reference photos (it often pays off to be neurotic) but not all of them, unfortunately. That said, searching for `172P` on http://airliners.net/ is a great way of getting reference pictures, albeit not close up ones.

> OK, I finished everything that was brought to my attention. 

@wlbragg I am out until the evening and then I will take a look at this and test it.

**Comment by gilbertohasnofb** [comment]
@wlbragg Managed to test it on my break: it is looking really good. The antennas are much better, the normal map above the wings looks spot on, the nose is way better and the handle on the other side of the nose looks good as well. :clap: 

What's left?

**Comment by gilbertohasnofb** [comment]
Sorry, just saw this:

> OK, I finished everything that was brought to my attention. The only outstanding issue is the reflection maps and I really don't think there is anything we can do about them.
>
> I'd like someone to verify that what is seen in the image I posted above is the same on other systems and that it is the same on other aircraft and not just ours. I'm talking about the reflection image in the glass being darker in the sloped front and back portions of the glass and lighter in the more vertical side glass.
>
> If that is true for other aircraft then it is an effect issue and not specific to the c172p. I verified it to be the same in at east a couple other aircraft but would like a couple more opinions from other's observations.

If nobody beats me to it I will test this in the evening. Then if the same is present in all other aircraft I will report back and merge this, else we wait before merging and try to fix it. Sounds good?

**Comment by wlbragg** [comment]
@gilbertohasnofb this last commit was a very minor position adjustment of the port side VHF antenna and cowling handle. Almost imperceivable adjustment. I noticed they might need moved ever so slightly when I was looking at @legoboyvdlp last image at the bottom of issue #1027

**Comment by wlbragg** [comment]
Cap livery...
cap1
cap2
cap3

**Comment by gilbertohasnofb** [comment]
@wlbragg This looks really good, thanks for the fixes. But the propeller (engine off) and the prop disk have both disappeared since my last pull, could you double check what happened?

**Comment by wlbragg** [comment]
@gilbertohasnofb 
I went ahead and split the windshield so you or @onox can merge this PR. I think we have to do the glass like this or the rain and ice effects look really terrible. So once this issue is decided I think we're done here.

**Comment by gilbertohasnofb** [comment]
@wlbragg Great, I will check this tonight and then merge this PR.

**Comment by gilbertohasnofb** [comment]
@wlbragg the split glass looks fine and I just pushed a texture for the nav light. Would you like to consider doing https://github.com/c172p-team/c172p-detailed/issues/1031 or should we just merge this and leave that for some other time? I'm fine either way. But if you want to tackle it, I can push another texture for the beacon light so that all we need to do is add a new "sphere" to the wing tip lights (no need to remodel anything else).

**Comment by gilbertohasnofb** [comment]
@wlbragg done, all liveries have the new lights now. Just missing that normal map issue (looks to me like the normal map of the flap is applied to the lights). Once that's done we are ready to merge.

**Comment by wlbragg** [comment]
OK, I think we're done here!

**Comment by gilbertohasnofb** [comment]
@wlbragg Fantastic work, as always. Thanks for the patience and sorry for pushing you a bit.

On the bright side, do compare this:



With this:

**Comment by wlbragg** [comment]
Yes, it has come a long way. You ready to come in out of the weather and start on the interior?

**Comment by gilbertohasnofb** [comment]
Absolutely! :smile: :+1: 

The main thing we should start with in my opinion is finding a new position for the panel itself. Then once that is set we can start altering it's model and adjusting/tweaking everything else. What do you think?

The main issue is the height and angle of the panel. The best references for that are external shots like the ones below. Notice how the cockpit line (yellow) is way higher than the highest point of the fuselage (green) and also much higher than the handles (purple), and therefore those last two should not be visible at all. 

image

And we also should try to make the interior fuselage a bit slimmer when it comes to thickness. Perhaps this kind of shot can help a bit:

image
 
Compare it to this:

image

I reckon we need to halve the thickness, more or less. 

Anyway, let's start discussing over those issues when you are ready and then we can create a PR and continue from there.

**Comment by gilbertohasnofb** [comment]
Sorry just a couple more shots of how this aircraft was looking like immediately before this project started:







:open_mouth:

**Pull request merged**